student.cpp

```diff
@@ -0,0 +1,248 @@
+{
+    cout << "Bye!" << endl;
+}
+
+/**
+ * The purpose of this method is to get a student ID
+ *
+ * @return int A student ID
+ */
+int ibonwe::icecream::Student::getId()
+{
+    return id;
+}
+
+/**
+ * The purpose of this method is to set a student ID
+ *
+ * @param int aId A student ID
+ */
+void ibonwe::icecream::Student::setId(int aId)
+{
+    id = aId;
+}
+
+/**
+ * The purpose of this method is to get a student number
+ *
+ * @return int A student number
+ */
+int ibonwe::icecream::Student::getStudentNumber()
+{
+    return studentNumber;
+}
+
+/**
+ * The purpose of this method is to set a student number
+ *
+ * @param int aStudentNumber A student number
+ */
+void ibonwe::icecream::Student::setStudentNumber(int aStudentNumber)
+{
+    studentNumber = aStudentNumber;
+}
+
+/**
+ * The purpose of this method is to get a student name
+ *
+ * @return string A student name
+ */
+string ibonwe::icecream::Student::getName()
+{
+    return name;
+}
+
+/**
+ * The purpose of this method is to set a student name
+ *
+ * @param string aName A student name
+ */
+void ibonwe::icecream::Student::setName(string aName)
+{
+    name = aName;
+}
+
+/**
+ * The purpose of this method is to get a student surname
+ *
+ * @return string A student surname
+ */
+string ibonwe::icecream::Student::getSurname()
+{
+    return surname;
+}
+
+/**
+ * The purpose of this method is to set a student surname
+ *
+ * @param int aSurname A student surname
+ */
+void ibonwe::icecream::Student::setSurname(string aSurname)
+{
+    surname = aSurname;
+}
+
+/**
+ * The purpose of this method is to get a student diploma name
+ *
+ * @return string A student diploma name
+ */
+string ibonwe::icecream::Student::getDiploma()
+{
+    return diploma;
+}
+
+/**
+ * The purpose of this method is to set a student diploma name
+ *
+ * @param int aDiploma A student diploma name
+ */
+void ibonwe::icecream::Student::setDiploma(string aDiploma)
+{
+    diploma = aDiploma;
+}
+
+/**
+ * The purpose of this method is to get a student average mark
+ *
+ * @return double A student average mark
+ */
+double ibonwe::icecream::Student::getAverageMark()
+{
+    return averageMark;
+}
+
+/**
+ * The purpose of this method is to set a student average mark
+ *
+ * @param double aAverageMark A student average mark
+ */
+void ibonwe::icecream::Student::setAverageMark(double aAverageMark)
+{
+    averageMark = aAverageMark;
+}
+
+/**
+ * The purpose of this method is to calculate the student average mark
+ *
+ * @return double A student average mark
+ */
+double ibonwe::icecream::Student::calcAverage()
+{
+    double sum;
+
+    for (int i = 0; i <= sizeof(results); i++)
+    {
+        sum += results[i];
+    }
+
+    return sum / sizeof(results);
+}
+
+/**
+ * The purpose of this method is to determine whether a student has passed the diploma
+ *
+ * @return a bool A student passed or failed
+ */
+bool ibonwe::icecream::Student::pass()
+{
+    if (getAverageMark == 50 || getAverageMark > 50)
```

Comment:
getAverageMark is a method not an attribute so should say getAverageMark(). But only call this once and store in a variable then use the variable in the if statement. There is need to do the same calculation twice in the if statement.